File: hw01/code/src/uniquinet.py
class UniqueIntProcessor:
    @staticmethod
    def extract_unique_integers(input_file_path, output_file_path):
        # Create a set to store unique integers
        unique_numbers = set()

        # Open the input file and read through each line
        with open(input_file_path, 'r') as file:
            for line_number, line in enumerate(file, start=1):
                # Clean the line by stripping whitespace
                cleaned_line = line.strip()
                
                # Ensure the line is not empty and contains exactly one integer
                if not cleaned_line or len(cleaned_line.split()) != 1:
                    print(f"Line {line_number} skipped: '{line.strip()}' (Invalid format)")
                    continue
                
                try:
                    # Convert the cleaned line to an integer
                    number = int(cleaned_line)
                    unique_numbers.add(number)  # Add number to the set
                except ValueError:
                    print(f"Line {line_number} skipped: '{line.strip()}' (Not an integer)")
                    continue

        # Sort the unique integers
        sorted_numbers = sorted(unique_numbers)

        # Write the sorted unique integers to the output file
        with open(output_file_path, 'w') as output_file:
            for number in sorted_numbers:
                output_file.write(f"{number}\n")

        # Print a summary message
        print(f"Sorted unique numbers written to: {output_file_path}")
        print("Unique numbers:", sorted_numbers)

# Example usage
if __name__ == "__main__":
    input_path = '../../sample_input_for_students/sample_01.txt'
    output_path = '../../results_for_sample_inputs/sample_01.txt_result.txt'
    UniqueIntProcessor.extract_unique_integers(input_path, output_path)

File: hw01/code/src/uniquinet.js
const fs = require('fs');

class UniqueIntProcessor {
    static extractUniqueIntegers(inputFilePath, outputFilePath) {
        // Create a set to store unique integers
        const uniqueNumbers = new Set();

        // Read the input file
        fs.readFile(inputFilePath, 'utf8', (err, data) => {
            if (err) {
                console.error("Error reading the file:", err.message);
                return;
            }

            // Split the file content into lines
            const lines = data.split('\n');
            lines.forEach((line, lineNumber) => {
                const cleanedLine = line.trim();

                // Ensure the line is not empty and contains exactly one integer
                if (!cleanedLine || cleanedLine.split(/\s+/).length !== 1) {
                    console.log(`Line ${lineNumber + 1} skipped: '${line}' (Invalid format)`);
                    return;
                }

                // Try to convert the cleaned line to an integer
                const number = parseInt(cleanedLine, 10);
                if (isNaN(number)) {
                    console.log(`Line ${lineNumber + 1} skipped: '${line}' (Not an integer)`);
                } else {
                    uniqueNumbers.add(number); // Add number to the set
                }
            });

            // Sort the unique integers
            const sortedNumbers = Array.from(uniqueNumbers).sort((a, b) => a - b);

            // Write the sorted unique integers to the output file
            fs.writeFile(outputFilePath, sortedNumbers.join('\n'), (err) => {
                if (err) {
                    console.error("Error writing to the file:", err.message);
                    return;
                }

                // Print a summary message
                console.log(`Sorted unique numbers written to: ${outputFilePath}`);
                console.log("Unique numbers:", sortedNumbers);
            });
        });
    }
}

// Example usage
const inputPath = '../../sample_input_for_students/sample_01.txt';
const outputPath = '../../results_for_sample_inputs/sample_01.txt_result.txt';
UniqueIntProcessor.extractUniqueIntegers(inputPath, outputPath);